src/core/diagram/qgsstackedbardiagram.cpp

```diff
@@ -117,7 +119,7 @@ double QgsStackedBarDiagram::legendSize( double value, const QgsDiagramSettings
 
 QString QgsStackedBarDiagram::diagramName() const
 {
-  return DIAGRAM_NAME_STACKED;
+  return QgsStackedBarDiagram::DIAGRAM_NAME_STACKED_BAR;
```

Comment:
@gacarrillor this has broken restoring old projects using stacked bar diagrams -- you'll need to revert this change, and then introduce a new unique, different string for the stacked diagram type.

Reply:
Thanks for the report and sorry.
Solved in #58915